test/e2e/wait_timeout_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestWaitTimeout(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+
+	yaml1 := `
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns1
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: ns1
+  name: simple-app
+spec:
+  selector:
+    matchLabels:
+      simple-app: ""
+  template:
+    metadata:
+      labels:
+        simple-app: ""
+    spec:
+      containers:
+        - name: simple-app
+          image: docker.io/dkalinin/k8s-simple-app@sha256:4c8b96d4fffdfae29258d94a22ae4ad1fe36139d47288b8960d9958d1e63a9d0
+          env:
+            - name: HELLO_MSG
+              value: stranger
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns2
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns3
```

Comment:
is there a reason why we need namespaces? can we just be using default default test namespace that kapp.Deploy automatically sets.

Reply:
No I was just creating that much namespaces such that Deployments will wait for the namespaces to get created and the changes will not get finished instantly and I can provide resource time value and make it fail. 

Yeah we can just use default namespace